prime-dotnet-webapi/Services/EnrolleeService.cs

```diff
@@ -1055,6 +1068,86 @@ public async Task<IEnumerable<HpdidLookup>> HpdidLookupAsync(IEnumerable<string>
                 .ToListAsync();
         }
 
+        public async Task<EnrolleeLookup> GpidLookupAsync(GpidLookupOptions option)
+        {
+            var careSettingIds = new List<int>();
+            var haIds = new List<int>();
+
+            switch (option.CareSetting)
+            {
+                case ProvisionerCareSettingCode.CommunityPharmacy:
+                    careSettingIds.Add((int)CareSettingType.CommunityPharmacy);
+                    break;
+                case ProvisionerCareSettingCode.PrivateCommunityHealthPractice:
+                    careSettingIds.Add((int)CareSettingType.CommunityPractice);
+                    break;
+                case ProvisionerCareSettingCode.FraserHealthAuthority:
+                    haIds.Add((int)HealthAuthorityCode.FraserHealth);
+                    break;
+                case ProvisionerCareSettingCode.InteriorHealthAuthority:
+                    haIds.Add((int)HealthAuthorityCode.InteriorHealth);
+                    break;
+                case ProvisionerCareSettingCode.VancouverCoastalHealthAuthority:
+                    haIds.Add((int)HealthAuthorityCode.VancouverCoastalHealth);
+                    break;
+                case ProvisionerCareSettingCode.VancouverIslandHealthAuthority:
+                    haIds.Add((int)HealthAuthorityCode.VancouverCoastalHealth);
+                    break;
+                case ProvisionerCareSettingCode.NorthernHealthAuthority:
+                    haIds.Add((int)HealthAuthorityCode.NorthernHealth);
+                    break;
+            }
+
+            return await _context.Enrollees
+                .Where(e => e.GPID == option.Gpid && e.FirstName == option.FirstName && e.LastName == option.LastName
+                    && e.CurrentStatus.StatusCode != (int)StatusType.Declined)
+                .Where(e => careSettingIds.Count() == 0 ||
+                    e.EnrolleeCareSettings.Where(s => careSettingIds.Contains(s.CareSettingCode))
+                        .Where(s => s.ConsentForAutoPull).Any())
+                .Where(e => haIds.Count() == 0 ||
+                    e.EnrolleeHealthAuthorities.Where(ha => haIds.Contains((int)ha.HealthAuthorityCode))
+                        .Where(ha => ha.ConsentForAutoPull).Any())
```

Comment:
Test scenario of Multiple HA care settings, but consent only for one of them doesn't appear to work.  E.g. enrollee is at both FHA and PHSA but only provides consent at FHA.  Requests for PHSA shouldn't return information

Reply:
I just tested locally and it works as design. Can you please test again?

Reply:
this is fixed